### DISCUSSION TITLE
Leaflet not executing in Filament Widget

### DISCUSSION THREAD

**Discussion by wpas81** [Leaflet/Leaflet#9170] [Q&A]
Hi

also posted here: https://stackoverflow.com/questions/77495795/leaflet-js-widget-in-filament-wont-execute 

For a non-profit I'm creating a Filament admin panel with a widget running a Leaflet.js map.

I've set this up according to the [Filemant docs](https://filamentphp.com/docs/3.x/support/assets#asynchronous-alpinejs-components)

All goes well, the data gets passed to the JS. But leaflet doesn't create the map. It seems that the L.map() command can't find the DIV I'm referrencing.

i've tried hardcoding it as const map = L.map('map'), L.map(document.getElementById('map')); L.map(this.$refs.leafletMap), L.map(this.$refs.leafletMap.id), ... none does the tric

Any help is hugely appreciated :)

**AdminPanelProvider.php**

```
class AdminPanelProvider extends PanelProvider
{
    public function boot()
    {
        FilamentAsset::register([
            Css::make('leaflet-1-9-4-css', 'https://unpkg.com/leaflet@1.9.4/dist/leaflet.css'),
            Js::make('leaflet-1-9-4-js', 'https://unpkg.com/leaflet@1.9.4/dist/leaflet.js'),
            AlpineComponent::make('visitor-heatmap-js', __DIR__ . '/../../../resources/js/leaflet-heatmap.js'),         
        ]);
    }
    
   ..
}
```
**visitor-heatmap-widget.blade**

```
<x-filament-widgets::widget>
    <x-filament::section>
    HEATMAP
        
        
        <div
            x-ignore
            ax-load
            ax-load-src="{{ \Filament\Support\Facades\FilamentAsset::getAlpineComponentSrc('visitor-heatmap-js') }}"
            x-data="leafletVisitorsHeatmap({
                cities: @js($data)
            })"
        >
            
                <div x-ref="leafletMap" id="map" style="width: 100%; height: 100%">

                </div>
            
        </div>
    </x-filament::section>
</x-filament-widgets::widget>
```
**leaflet-heatmap.js (stripped to the bare minimum)**

```
export default function leafletVisitorsHeatmap({ cities }) 
{
    return {
 
        init: function () {
            console.log('start leafletVisitorsHeatmap');
            //console.log('cities:');
            //console.log(cities);
                    
            const map = L.map(this.$refs.leafletMap.id).setView([51.505, -0.09], 13);

            const tiles = L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
                maxZoom: 19,
                attribution: '&copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>'
            }).addTo(map);
            
            
        },
        
    }
}
```

**Comment by Falke-Design** [comment]
Hi, and thanks for taking the time to open a report in Leaflet.

However, in this repository we can only handle issues in "[vanilla](http://vanilla-js.com/)" Leaflet. This means that we do not handle issues which are specific to frameworks such as:

    React
    Ractive
    AngularJS
    Bootstrap
    Leaflet for R
    Joomla
    Wordpress
    Ionic
    Cordova
    Vue
    Electron
    Polymer
    TypeScript
    D3
    Folium
    NativeScript
    ...etc

Please understand that we only have the time and energy to test Leaflet in plain web browsers in a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example).
